pkg/getter/ocigetter.go

```diff
@@ -50,10 +52,28 @@ func (g *OCIGetter) get(href string) (*bytes.Buffer, error) {
 			registry.PullOptWithProv(true))
 	}
 
-	if version := g.opts.version; version != "" {
-		ref = fmt.Sprintf("%s:%s", ref, version)
+	// Retrieve list of repository tags
+	tags, err := client.Tags(ref)
+	if err != nil {
+		return nil, err
+	}
+	if len(tags) == 0 {
+		return nil, errors.Errorf("Unable to locate any tags in provided repository: %s", ref)
+	}
+
+	// Determine if version provided
+	// If empty, try to get the highest available tag
+	// If exact version, try to find it
+	// If semver constraint string, try to find a match
+	providedVersion := g.opts.version
+
+	tag, err := registry.GetTagMatchingVersionOrConstraint(tags, providedVersion)
+	if err != nil {
+		return nil, err
 	}
 
+	ref = fmt.Sprintf("%s:%s", ref, tag)
```

Comment:
Sorting out the version for the other downloaders happens in another place. In `DownloadTo` a version or range is passed. The first thing `DownloadTo` does is call `c.ResolveChartVersion` which turns a version into a URI to call. This is then passed off to a getter which does the thing it was designed to do... get the thing.

Shouldn't this getter function using a similar code flow so the logic is in the same places?

Reply:
done